src/main/java/com/redhat/cloud/event/parser/ConsoleCloudEventParser.java

```diff
@@ -96,8 +96,13 @@ public <T extends GenericConsoleCloudEvent<?>> T fromJsonString(String cloudEven
             // Verify it's a valid Json
             JsonNode cloudEvent = objectMapper.readTree(cloudEventJson);
             validate(cloudEvent, jsonSchema);
-
-            return objectMapper.treeToValue(cloudEvent, consoleCloudEventClass);
+            if(consoleCloudEventClass.isInstance(ConsoleCloudEvent.class)) {
+                ConsoleCloudEvent consoleCloudEvent = objectMapper.treeToValue(cloudEvent, ConsoleCloudEvent.class);
+                consoleCloudEvent.setObjectMapper(this.objectMapper);
+                return (T) consoleCloudEvent;
+            } else {
+                return objectMapper.treeToValue(cloudEvent, consoleCloudEventClass);
+            }
```

Comment:
Good catch! although I would prefer something like (if it is valid code :sweat_smile: ):

```suggestion
            T genericCloudEvent = objectMapper.treeToValue(cloudEvent, consoleCloudEventClass);
            if (genericCloudEvent instanceof ConsoleCloudEvent) {
                ConsoleCloudEvent consoleCloudEvent = (ConsoleCloudEvent) genericCloudEvent;
                consoleCloudEvent.setObjectMapper(this.objectMapper);
            }

            return genericCloudEvent;
```

Reply:
Sure, your way looks better :)